pgoapi/pgoapi.py

```diff
@@ -439,31 +441,39 @@ def cleanup_pokemon(self, inventory_items=None):
             if len(pokemons) > self.MIN_SIMILAR_POKEMON:
                 # highest CP pokemon first
                 pokemons = sorted(pokemons, key=lambda pok: (pok.cp, pok.iv), reverse=True)
-                pokemons_keep_counter = 0
-                pokemons_keep_iv_counter = 0
+                self._pokemons_keep_counter = 0
+                self._pokemons_keep_iv_counter = 0
 
                 for pokemon in pokemons:
-                    # never release favorites and other defined pokemons
-                    if pokemon.is_favorite or pokemon.pokemon_id in self.keep_pokemon_ids:
-                        pokemons_keep_counter += 1
-                        continue
-                    # release defined throwaway pokemons
-                    elif pokemon.pokemon_id in self.throw_pokemon_ids:
-                        self.do_release_pokemon(pokemon)
-                    # keep high-iv pokemons based on config values
-                    elif pokemon.iv > self.KEEP_IV_OVER and pokemons_keep_iv_counter < self.MAX_POKEMON_HIGH_IV and \
-                                    pokemon.cp > (pokemons[0].cp * self.KEEP_IV_MIN_PERCENT_CP / 100):
-                        pokemons_keep_iv_counter += 1
-                        pokemons_keep_counter += 1
-                        continue
-                    # keep high-cp pokemons and first MIN_SIMILAR_POKEMON amount of pokemons
-                    elif pokemon.cp > self.KEEP_CP_OVER or pokemons_keep_counter < self.MIN_SIMILAR_POKEMON:
-                        pokemons_keep_counter += 1
-                        continue
-                    # release all other pokemons
-                    else:
+                    if self.is_pokemon_eligible_for_transfer(pokemon, pokemons[0]):
                         self.do_release_pokemon(pokemon)
 
+                # reset the counters just in case
+                self._pokemons_keep_iv_counter = 0
+                self._pokemons_keep_counter = 0
+
+    def is_pokemon_eligible_for_transfer(self, pokemon, best_pokemon=None):
```

Comment:
instead of self._pokemons_keep_iv_counter just maintain it in the loop
like 
```
keep_iv_counter = 0
keep_pokemon_counter =0
for pokemon in pokemons:
    (iv, keep) = is_pokemon_eligible_for_transfer(pokemon, best_pokemon, keep_pokemon_counter, keep_iv_counter)
    if iv:
        keep_iv_counter += 1
    if keep:
        keep_pokemon_counter += 1
```
 and you can return (True, False) depending on what counter you want to increment

This way we do not add more attributes to the class itself since its really not a part of the class. 
Let me know if that makes sense